inject/src/main/java/io/micronaut/inject/visitor/VisitorContext.java

```diff
@@ -99,6 +104,39 @@
         return Collections.emptyList();
     }
 
+    /**
+     * Obtain the project directory.
+     *
+     * @return An optional wrapping the project directory
+     */
+    default Optional<Path> projectDir() {
```

Comment:
Rename to `getProjectDir()` also should be implemented in the child implementations so that the computed `Path` is cached and the computation is not done each time this is called. This can be done by simply calling the super interface

Reply:
I don't think it is necessary to implement the method on the child implementations. The first thing we do in it is:
```java
Optional<Path> projectDir = get(MICRONAUT_PROJECT_DIR, Path.class);
if (projectDir.isPresent()) {
    return projectDir;
}
```
That will get the cached directory once it has been calculated.